gridscale/resource_gridscale_k8s.go

```diff
@@ -445,3 +408,60 @@ func getK8sTemplateUUID(client *gsclient.Client, release string) (string, error)
 
 	return uTemplate.Properties.ObjectUUID, nil
 }
+
+func validateK8sParameters(d *schema.ResourceDiff, template gsclient.PaaSTemplate) error {
+	var errorMessages []string
+	worker_memory_scheme, mem_ok := template.Properties.ParametersSchema["k8s_worker_node_ram"]
+	// TODO: The API scheme will be CHANGED in the future. There will be multiple node pools.
+	if memory, ok := d.GetOk("node_pool.0.memory"); ok && mem_ok {
+		if worker_memory_scheme.Min > memory.(int) || memory.(int) > worker_memory_scheme.Max {
+			errorMessages = append(errorMessages, fmt.Sprintf("Invalid 'node_pool.0.memory' value. Value must stays between %d and %d\n", worker_memory_scheme.Min, worker_memory_scheme.Max))
+		}
+	}
+
+	worker_core_scheme, core_ok := template.Properties.ParametersSchema["k8s_worker_node_cores"]
+	// TODO: The API scheme will be CHANGED in the future. There will be multiple node pools.
+	if core, ok := d.GetOk("node_pool.0.cores"); ok && core_ok {
+		if worker_core_scheme.Min > core.(int) || core.(int) > worker_core_scheme.Max {
+			errorMessages = append(errorMessages, fmt.Sprintf("Invalid 'node_pool.0.cores' value. Value must stays between %d and %d\n", worker_core_scheme.Min, worker_core_scheme.Max))
+		}
+	}
+
+	worker_count_scheme, worker_count_ok := template.Properties.ParametersSchema["k8s_worker_node_count"]
+	// TODO: The API scheme will be CHANGED in the future. There will be multiple node pools.
+	if node_count, ok := d.GetOk("node_pool.0.node_count"); ok && worker_count_ok {
+		if worker_count_scheme.Min > node_count.(int) || node_count.(int) > worker_count_scheme.Max {
+			errorMessages = append(errorMessages, fmt.Sprintf("Invalid 'node_pool.0.node_count' value. Value must stays between %d and %d\n", worker_count_scheme.Min, worker_count_scheme.Max))
+		}
+	}
+
+	worker_storage_scheme, storage_ok := template.Properties.ParametersSchema["k8s_worker_node_storage"]
+	// TODO: The API scheme will be CHANGED in the future. There will be multiple node pools.
+	if storage, ok := d.GetOk("node_pool.0.storage"); ok && storage_ok {
+		if worker_storage_scheme.Min > storage.(int) || storage.(int) > worker_storage_scheme.Max {
```

Comment:
I would assume the variable is always on the left side and the value on the right. This becomes ```storage.(int) > worker_storage_scheme.Min   || storage.(int) > worker_storage_scheme.Max```